polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -231,21 +223,25 @@ function IndexTableBase({
 
   const [canScrollRight, setCanScrollRight] = useState(true);
 
-  const handleCanScrollRight = useCallback(() => {
-    if (
-      !lastColumnSticky ||
-      !tableElement.current ||
-      !scrollableContainerElement.current
-    ) {
-      return;
-    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm confused by this, why not pass an inline function that returns the `debounce`? I see it done in both ways in this file, what's the difference?

Reply:
ah I get it, the difference between this executing immediately vs calling it directly.